x-pack/plugins/cloud_security_posture/public/components/fleet_extensions/gcp_credential_form.tsx

```diff
@@ -240,56 +251,76 @@ const GcpInputVarFields = ({
   fields: Array<GcpFields[keyof GcpFields] & { value: string; id: string }>;
   onChange: (key: string, value: string) => void;
 }) => {
-  const [credentialOption, setCredentialOption] = useState('Credentials File');
-  const targetFieldName = (id: string) => {
+  const getFieldById = (id: keyof GcpInputFields['fields']) => {
     return fields.find((element) => element.id === id);
   };
+  const [credentialOption, setCredentialOption] = useState(credentialOptionsList[0].value);
+  const credentialsTypeField = {
+    id: 'credentials_type',
+    label: 'Credentials Type',
+    type: 'text',
+    value: credentialOption,
+  };
+  const projectIdFields = getFieldById('project_id');
+  const credentialsTypeFields = getFieldById('credentials_type');
+  const credentialFilesFields = getFieldById('credentials_file');
+  const credentialJSONFields = getFieldById('credentials_json');
+
+  const credentialFieldValue = credentialOptionsList[0].value;
+  const credentialJSONValue = credentialOptionsList[1].value;
```

Comment:
i guess its a nit but a best practice would be to set this value using a constant and then use it both when setting the options and when using it to conditionally render fields.